src/components/v5/common/Fields/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,180 @@
+
+import styles from './Datepicker.css';
+
+const Datepicker = <WithRange extends boolean = false>({
+  cancelButtonProps,
+  applyButtonProps,
+  popperModifiers,
+  selectsRange,
+  onChange,
+  startDate: startDateProp,
+  endDate: endDateProp,
+  selected: selectedProp,
+  dateFormat = DEFAULT_DATE_FORMAT,
+  popperClassName,
+  ...rest
+}: DatepickerProps<WithRange>): ReturnType<React.FC> => {
+  const isMobile = useMobile();
+  const calendarRef = useRef<DatePicker<never, WithRange>>(null);
+
+  const [startDate, setStartDate] = useState<Date | null>(
+    startDateProp || null,
+  );
+  const [endDate, setEndDate] = useState<Date | null>(endDateProp || null);
+
+  const handleChange: DatepickerProps<WithRange>['onChange'] = useCallback(
+    (dates) => {
+      if (!dates && selectsRange) {
+        setStartDate(null);
+        setEndDate(null);
+        return;
+      }
+
+      if (Array.isArray(dates)) {
+        const [start, end] = dates;
+        setStartDate(start);
+        setEndDate(end);
+      } else {
+        setStartDate(dates);
+      }
+    },
+    [selectsRange],
+  );
+
+  const resetValues = useCallback(() => {
+    if (selectsRange) {
+      setStartDate(startDateProp || null);
+      setEndDate(endDateProp || null);
+    } else {
+      setStartDate(selectedProp || null);
+      setEndDate(null);
+    }
+  }, [endDateProp, selectedProp, selectsRange, startDateProp]);
+
+  const {
+    onClick: applyButtonOnClick,
+    mode: applyButtonMode,
+    text: applyButtonText,
+  } = applyButtonProps || {};
+  const {
+    onClick: cancelButtonOnClick,
+    mode: cancelButtonMode,
+    text: cancelButtonText,
+  } = cancelButtonProps || {};
+
+  return (
+    <DatePicker<never, WithRange>
+      portalId={isMobile ? DATEPICKER_PORTAL_ID : undefined}
+      calendarContainer={DatepickerContainer}
+      calendarClassName={styles.wrapper}
+      renderCustomHeader={(props) => (
+        <DatepickerCustomHeader
+          startDate={startDate}
+          endDate={selectsRange ? endDate : null}
+          dateFormat={dateFormat}
+          setStartDate={setStartDate}
+          setEndDate={selectsRange ? setEndDate : undefined}
+          onClose={
+            isMobile
+              ? () => {
+                  resetValues();
+                  calendarRef.current?.setOpen(false);
+                }
+              : undefined
+          }
+          {...props}
+        />
+      )}
+      ref={calendarRef}
+      customInput={
+        selectsRange ? (
+          <DatepickerCustomRangeInput
+            startDate={startDateProp}
+            endDate={endDateProp}
+            dateFormat={dateFormat}
+          />
+        ) : (
+          <DatepickerCustomInput />
+        )
+      }
+      dateFormat={dateFormat}
+      popperClassName={clsx(popperClassName, 'max-w-[20.5rem] !z-[1000]')}
+      renderDayContents={(day) => (
+        <div className="react-datepicker__day-content">{day}</div>
+      )}
```

Comment:
I'm really not a fan of mixing global classes and tailwind classes

Reply:
I'm using this class with CSS Modules. With tailwind classes, it would be pretty hard to modify some elements based on other classes added to the Datepicker by default.